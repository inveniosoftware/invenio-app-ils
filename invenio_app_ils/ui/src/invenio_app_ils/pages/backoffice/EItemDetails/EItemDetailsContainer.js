import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import { EItemDetails } from './components';

export default class EItemDetailsContainer extends Component {
  constructor(props) {
    super(props);
    this.fetchEItemDetails = this.props.fetchEItemDetails;
  }

  componentDidMount() {
    this.fetchEItemDetails(this.props.match.params.eitemPid);
  }

  render() {
    return (
      <Container>
        <EItemDetails />
      </Container>
    );
  }
}

EItemDetailsContainer.propTypes = {
  fetchEItemDetails: PropTypes.func.isRequired,
};
