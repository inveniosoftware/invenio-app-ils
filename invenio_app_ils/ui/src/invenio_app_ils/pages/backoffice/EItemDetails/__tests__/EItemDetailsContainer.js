import React from 'react';
import { shallow, mount } from 'enzyme';
import EItemDetailsContainer from '../EItemDetailsContainer';

jest.mock('../../../../common/config');

jest.mock('../components/EItemDetails', () => {
  return {
    EItemDetails: () => null,
  };
});

describe('EItemDetailsContainer tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const routerUrlParams = {
    params: {
      eitemPid: 111,
    },
  };

  it('should load the details component', () => {
    const component = shallow(
      <EItemDetailsContainer
        match={routerUrlParams}
        fetchEItemDetails={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch eitem details on mount', () => {
    const mockedFetchEItemDetails = jest.fn();
    component = mount(
      <EItemDetailsContainer
        match={routerUrlParams}
        fetchEItemDetails={mockedFetchEItemDetails}
      />
    );
    expect(mockedFetchEItemDetails).toHaveBeenCalledWith(
      routerUrlParams.params.eitemPid
    );
  });
});
