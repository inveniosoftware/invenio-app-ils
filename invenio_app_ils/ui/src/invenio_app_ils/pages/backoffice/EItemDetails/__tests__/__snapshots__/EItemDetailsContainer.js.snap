// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EItemDetailsContainer tests should load the details component 1`] = `
<Container>
  <EItemDetails />
</Container>
`;
